ostd/src/task/mod.rs

```diff
@@ -55,6 +56,10 @@ pub struct Task {
     #[expect(dead_code)]
     kstack: KernelStack,
 
+    /// This is to check if we would context switch to a running task.
+    // TODO: use `#[cfg(debug_assertions)]` when the scheduler API is sound.
```

Comment:
This comment implies that this check can be skipped for release builds. This is incorrect. Our safety guarantee must be held for both debug and release builds. And the soundness of our scheduler injection API relies on this check.